ambry-frontend/src/test/java/com.github.ambry.frontend/AmbrySecurityServiceTest.java

```diff
@@ -123,119 +125,110 @@ public void testProcessRequest()
-        null);
-    callback.reset();
-    restResponseChannel = new MockRestResponseChannel();
-    restRequest = createRestRequest(RestMethod.HEAD, "/", null);
-    securityService.processResponse(restRequest, restResponseChannel, blobInfo, callback).get();
-    Assert.assertTrue("Call back should have been invoked", callback.callbackInvoked.get());
-    Assert.assertNull("Exception should not have been thrown", callback.exception);
-    Assert.assertEquals("No body is expected in the response", 0, restResponseChannel.getResponseBody().length);
-    verifyHeadersForHead(blobInfo.getBlobProperties(), restResponseChannel);
-
-    // Get Blobinfo
-    callback.reset();
-    restResponseChannel = new MockRestResponseChannel();
-    restRequest = createRestRequest(RestMethod.GET, "/abc/" + RestUtils.SubResource.BlobInfo, null);
-    securityService.processResponse(restRequest, restResponseChannel, blobInfo, callback).get();
-    Assert.assertTrue("Call back should have been invoked", callback.callbackInvoked.get());
-    Assert.assertNull("Exception should not have been thrown", callback.exception);
-    Assert.assertEquals("Response should have been set ", ResponseStatus.Ok, restResponseChannel.getStatus());
-    Assert.assertEquals("No body is expected in the response", 0, restResponseChannel.getResponseBody().length);
-    verifyBlobPropertiesHeaders(blobInfo.getBlobProperties(), restResponseChannel);
+    // with callbacks
+    // for unsupported methods
+    methods = new RestMethod[]{RestMethod.DELETE};
+    for (RestMethod restMethod : methods) {
+      testExceptionCasesProcessResponse(restMethod, new MockRestResponseChannel(), DEFAULT_INFO,
+          RestServiceErrorCode.InternalServerError);
+    }
 
-    // Get Blob
-    testGetBlob(callback, blobInfo);
-    testGetNotModifiedBlob(callback, blobInfo);
+    // HEAD
+    // normal
+    testHeadBlob(DEFAULT_INFO);
+    // with no owner id
+    BlobInfo blobInfo =
+        new BlobInfo(new BlobProperties(100, SERVICE_ID, null, "image/gif", false, Utils.Infinite_Time), null);
+    testHeadBlob(blobInfo);
+    // with no content type
+    blobInfo = new BlobInfo(new BlobProperties(100, SERVICE_ID, OWNER_ID, null, false, Utils.Infinite_Time), null);
+    testHeadBlob(blobInfo);
+    // with a TTL
+    blobInfo = new BlobInfo(new BlobProperties(100, SERVICE_ID, OWNER_ID, "image/gif", false, 10000), null);
+    testHeadBlob(blobInfo);
+
+    // GET BlobInfo
+    testGetSubResource(RestUtils.SubResource.BlobInfo);
+    // GET UserMetadata
+    testGetSubResource(RestUtils.SubResource.UserMetadata);
+
+    // POST
+    testPostBlob();
+
+    // GET Blob
+    // less than chunk threshold size
+    blobInfo = new BlobInfo(
+        new BlobProperties(FRONTEND_CONFIG.frontendChunkedGetResponseThresholdInBytes - 1, SERVICE_ID, OWNER_ID,
+            "image/gif", false, 10000), null);
+    testGetBlob(blobInfo);
+    // more than chunk threshold size
```

Comment:
why not a test where in blob size is equal to frontendChunkedGetResponseThresholdInBytes